tests/Unit/Cli/CampaignValidatorTest.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Tests\Unit\Cli;
+
+use WMDE\Fundraising\Frontend\BucketTesting\Bucket;
+use WMDE\Fundraising\Frontend\BucketTesting\Campaign;
+use WMDE\Fundraising\Frontend\BucketTesting\CampaignCollection;
+use WMDE\Fundraising\Frontend\Cli\CampaignConfigValidation\CampaignValidator;
+use WMDE\Fundraising\Frontend\Cli\CampaignConfigValidation\ValidationErrorLogger;
+use WMDE\Fundraising\Frontend\Cli\CampaignConfigValidation\Rule\DefaultBucketRule;
+use WMDE\Fundraising\Frontend\Cli\CampaignConfigValidation\Rule\StartAndEndTimeRule;
+use WMDE\Fundraising\Frontend\Cli\CampaignConfigValidation\Rule\UniqueBucketRule;
+use WMDE\Fundraising\Frontend\Tests\Fixtures\FakeBucket;
+
+/**
+ * @covers \WMDE\Fundraising\Frontend\Cli\CampaignConfigValidation\CampaignValidator
+ */
+class CampaignValidatorTest extends \PHPUnit\Framework\TestCase {
+
+	public function testWhenValidCampaignIsSupplied_validationPasses(): void {
+		$errorLogger = new ValidationErrorLogger();
+		$bucket = FakeBucket::createBucket();
+		$campaignCollection = new CampaignCollection( $bucket->getCampaign() );
```

Comment:
I had a look at `FakeBucket` and it creates a Campaign with one bucket. Let's discuss if that is something that's useful (and scenarios where we would use that) or we should add a validation rules for "campaign needs to have at least 2 buckets".